Purge() causes panic with TTL.

The following code panics:
```
package main

import (
	"time"

	"zvelo.io/ttlru"
)

func main() {
	cache := ttlru.New(1, ttlru.WithTTL(time.Millisecond*10))

	cache.Set("bug", "foo")
	cache.Purge()
	time.Sleep(time.Millisecond * 50)
	// panic above.
}
```
```
; go run main.go
panic: runtime error: index out of range

goroutine 17 [running]:
zvelo.io/ttlru.ttlHeap.Swap(0xc00000c020, 0x0, 0x1, 0x0, 0xffffffffffffffff)
        /home/dar/go/pkg/mod/zvelo.io/ttlru@v1.0.3/ttlheap.go:14 +0x86
container/heap.Remove(0x4bb0e0, 0xc00000a0c0, 0x0, 0xc000058060, 0xc0000367a8)
        /home/dar/.asdf/installs/golang/1.11.2/go/src/container/heap/heap.go:75 +0x9c
zvelo.io/ttlru.(*cache).removeEntry(0xc000058040, 0xc00006c000)
        /home/dar/go/pkg/mod/zvelo.io/ttlru@v1.0.3/ttlru.go:176 +0x7b
zvelo.io/ttlru.(*cache).insertEntry.func1()
        /home/dar/go/pkg/mod/zvelo.io/ttlru@v1.0.3/ttlru.go:137 +0x85
created by time.goFunc
        /home/dar/.asdf/installs/golang/1.11.2/go/src/time/sleep.go:172 +0x44
exit status 2
```

From the code it appears that Purge() simply overwrites the existing heap without modifying any existing entries.  When the TTL go routine for each entry kicks off, it tries to removeEntry itself from the heap, only its heap is no longer present, and its index refers to a slot that is no longer in the list.

--- a/ttlru.go
+++ b/ttlru.go
@@ -228,6 +228,10 @@ func (c *cache) Purge() {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
+	for _, e := range c.items {
+		e.index = -1
+	}
+
 	h := make(ttlHeap, 0, c.cap)
 	c.heap = &h
 	c.items = make(map[interface{}]*entry, c.cap)

